astropy/table/column.py

```diff
@@ -45,6 +45,130 @@ def _auto_names(n_cols):
+    # stored in _astropy_column_attrs.  For BaseColumn instances use the usual setattr.
+    if isinstance(col, BaseColumn):
+        setattr(col, attr, value)
+    else:
+        # If no _astropy_column_attrs or it is None then convert to dict
+        if getattr(col, '_astropy_column_attrs', None) is None:
+            col._astropy_column_attrs = {}
+        if attr == 'parent_table':
+            value = None if value is None else weakref.ref(value)
+        col._astropy_column_attrs[attr] = value
+
+def col_getattr(col, attr, default=None):
+    """
+    Get one of the column attributes
+
+    Warning: this function is subject to change or removal.
+    """
+    if attr not in COLUMN_ATTRS:
+        raise AttributeError("attribute must be one of {0}".format(COLUMN_ATTRS))
+
+    # The unit and dtype attributes are considered univeral and do NOT get
+    # stored in _astropy_column_attrs.  For BaseColumn instances use the usual setattr.
+    if (isinstance(col, BaseColumn) or
+            (isinstance(col, Quantity) and attr in ('dtype', 'unit'))):
+        value = getattr(col, attr, default)
+    else:
+        # If col does not have _astropy_column_attrs or it is None (meaning
+        # nothing has been set yet) then return default, otherwise look for
+        # the attribute in the astropy_column_attrs dict.
+        if getattr(col, '_astropy_column_attrs', None) is None:
+            value = default
+        else:
+            value = col._astropy_column_attrs.get(attr, default)
+
+        # Weak ref for parent table
+        if attr == 'parent_table' and callable(value):
+            value = value()
+
+        # Mixins have a default dtype of Object if nothing else was set
+        if attr == 'dtype' and value is None:
+            value = np.dtype('O')
+
+    return value
+
+def _col_update_attrs_from(newcol, col, exclude_attrs=['name', 'parent_table']):
+    """
+    Update _astropy_column_attrs from mixin `col` to `newcol`.  Does nothing
+    for BaseColumn cols
+
+    Warning: this function is subject to change or removal.
+    """
+    if isinstance(newcol, BaseColumn):
+        return
+
+    attrs = COLUMN_ATTRS - set(exclude_attrs)
+    for attr in attrs:
+        val = col_getattr(col, attr)
+        if val is not None:
+            col_setattr(newcol, attr, deepcopy(val))
+
+def col_iter_str_vals(col):
+    """
+    This is a mixin-safe version of Column.iter_str_vals.
+
+    Warning: this function is subject to change or removal.
+    """
+    parent_table = col_getattr(col, 'parent_table')
+    formatter = FORMATTER if parent_table is None else parent_table.formatter
+    _pformat_col_iter = formatter._pformat_col_iter
+    for str_val in _pformat_col_iter(col, -1, False, False, {}):
+        yield str_val
+
+def col_copy(col):
+    """
+    This is a mixin-safe version of Column.copy() (with copy_data=True).
+
+    Warning: this function is subject to change or removal.
```

Comment:
Perhaps this should be an actual ``.. warning::`` so that it appears more alarming in the sphinx docs?